rfd/0149-headless-kube.md

```diff
@@ -0,0 +1,242 @@
+---
+authors:  Anton Miniailo (anton@goteleport.com)
+state: Draft
+---
+
+# RFD 0149 - Headless Kubernetes Access
+
+## Required Approvals
+
+* Engineering: @r0mant && @tigrato
+* Product: @xinding33 || @klizhentas
+* Security: @reedloden || @jentfoo
+
+# What
+
+Support headless authentication for Teleport Kubernetes access.
+Headless authentication means that a user will be able to use their credentials and an MFA device on a local machine to authenticate
+on a remote machine.
+
+# Why
+
+Some users have a workflow whereby they do a big part of their work on a remote Linux machine, to which they connect by SSH from their
+local machine. If they access Teleport through SSO they encounter a problem - they don't have access to the browser in the terminal
+on that remote machine, so they can't authenticate. Authentication on a remote machine using SSO/MFA device on the local machine
+unlocks such workflows. This was already implemented for `tsh ssh` access, but is not yet available for Kubernetes access.
+
+# Scope
+
+The main goal of this proposal is to provide headless Kubernetes access for remote Linux machines, as it addresses the
+majority of use cases requiring this feature. MacOS might be supported with a slightly different approach,
+without using a memory-only kubeconfig file.
+
+# Details
+
+To implement headless Kubernetes access, we'll build upon two recently implemented features -
+headless authentication for `tsh ssh` ([RFD 105](https://github.com/gravitational/teleport/blob/master/rfd/0105-headless-authentication.md)) and Teleport Kubernetes
+local proxy ([RFD 121](https://github.com/gravitational/teleport/blob/master/rfd/0121-kube-mfa-sessions.md)).
+
+## Headless authentication
+
+When headless authentication is initiated by the user on a remote machine, the client calls the `POST /webapi/login/headless`
+endpoint on the Auth server. It creates a `HeadlessAuthentication` resource on the backend.
+After that, the call is paused, and the user is provided with a link (or command line alternative) where they can approve
+the newly created headless authentication request using their local machine with a logged-in Teleport and an MFA device. Once
+the request is approved by the user, the authentication process is unpaused and the original call returns
+certificates created by the auth server. These certificates have a TTL of 1 minute.
```

Comment:
Which certificates?

Reply:
This is regular Teleport login certificates. I added it to the text.